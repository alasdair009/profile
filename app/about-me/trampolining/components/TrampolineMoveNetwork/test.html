<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
