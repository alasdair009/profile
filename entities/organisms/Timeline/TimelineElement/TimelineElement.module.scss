.logo {
  left: 50%;
  max-height: 75%;
  max-width: 75%;
  object-fit: contain;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
}
