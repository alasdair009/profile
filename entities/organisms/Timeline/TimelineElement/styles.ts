import styled from "styled-components";
import Image from "next/image";

export const Root = styled.div``;

export const Logo = styled(Image)`
  left: 50%;
  max-height: 75%;
  max-width: 75%;
  object-fit: contain;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
`;
