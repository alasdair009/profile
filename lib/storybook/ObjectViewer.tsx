import { Source } from "@storybook/blocks";

import type { JSX } from "react";

export type ObjectViewerProps = {
  /**
   * The object to display
   */
  object: object;
  /**
   * The title of the code segment
   */
  title: string;
};

export function ObjectViewer({
  object,
  title,
}: ObjectViewerProps): JSX.Element {
  // Format the object as JSON with indentation
  const objectAsString = JSON.stringify(object, null, 4);

  return (
    <div title={title} data-testid={ObjectViewer.name}>
      <Source code={objectAsString} dark={true} />
    </div>
  );
}
