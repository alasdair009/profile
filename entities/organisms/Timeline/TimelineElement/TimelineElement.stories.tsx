import { TimelineElement } from "./TimelineElement";
import { Meta, StoryObj } from "@storybook/react";
import jagexLogo from "../../../assets/jagex-logo.svg";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof TimelineElement> = {
  component: TimelineElement,
  args: {
    date: "2011 - present",
    icon: jagexLogo,
    alt: "first logo",
    children: "abc",
  },
};
export default meta;

export const Default: StoryObj<typeof TimelineElement> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const timelineElement = canvas.getByTestId(TimelineElement.name);

    await expect(timelineElement).toBeInTheDocument();
    await expect(canvas.getByText(args.date)).toBeInTheDocument();
    await expect(canvas.getByAltText(args.alt)).toBeInTheDocument();
    await expect(canvas.getByText(args.children as string)).toBeInTheDocument();
  },
};
