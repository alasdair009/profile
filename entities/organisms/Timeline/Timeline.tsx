"use client";
import { Root } from "./styles";
import { HTMLAttributes, ReactNode } from "react";
import { VerticalTimeline } from "react-vertical-timeline-component";
import "react-vertical-timeline-component/style.min.css";
import { StaticImageData } from "next/image";
import { TimelineElement } from "./TimelineElement";
import { prefersReducedMotion } from "@/entities";

export type TimelineEntry = {
  date: string;
  icon: StaticImageData;
  alt: string;
  content: ReactNode;
};

export type TimelineProps = {
  entries: TimelineEntry[];
} & HTMLAttributes<HTMLDivElement>;

export function Timeline({ entries, ...rest }: TimelineProps) {
  return (
    <Root data-testid={Timeline.name} {...rest}>
      <VerticalTimeline animate={!prefersReducedMotion}>
        {entries.map((entry, index) => (
          <TimelineElement
            key={`${entry.date}${entry.icon}${index}`}
            date={entry.date}
            icon={entry.icon}
            alt={entry.alt}
          >
            {entry.content}
          </TimelineElement>
        ))}
      </VerticalTimeline>
    </Root>
  );
}
