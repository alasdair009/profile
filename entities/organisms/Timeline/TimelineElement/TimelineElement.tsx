"use client";
import { StaticImageData } from "next/image";
import { colors, sizes } from "@/entities";
import { Logo, Root } from "./styles";
import { VerticalTimelineElement } from "react-vertical-timeline-component";
import { useInView } from "react-intersection-observer";
import { HTMLAttributes } from "react";
import "react-vertical-timeline-component/style.min.css";

type TimelineElementProps = {
  date: string;
  icon: StaticImageData;
  alt: string;
} & HTMLAttributes<HTMLDivElement>;

export function TimelineElement({
  date,
  icon,
  alt,
  children,
  ...rest
}: TimelineElementProps) {
  const { ref, inView } = useInView({
    triggerOnce: true,
  });
  return (
    <Root
      ref={ref}
      className="vertical-timeline-element"
      data-testid={TimelineElement.name}
      {...rest}
    >
      <VerticalTimelineElement
        visible={inView}
        contentStyle={{
          background: colors.greenGrass,
          color: colors.whiteGhost,
        }}
        contentArrowStyle={{
          borderRight: `${sizes.s8.rem} solid  ${colors.greenGrass}`,
        }}
        date={date}
        iconStyle={{
          background: colors.greenGrass,
          color: colors.whiteGhost,
        }}
        icon={<Logo src={icon} alt={alt} />}
      >
        {children}
      </VerticalTimelineElement>
    </Root>
  );
}
