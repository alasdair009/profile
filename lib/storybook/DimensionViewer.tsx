import { Source } from "@storybook/blocks";
import type { HTMLAttributes } from "react";

import { ObjectViewer } from "./ObjectViewer";
import { colors } from "@/entities";

export type DimensionViewerProps = {
  /**
   * The units to display
   */
  units: Record<string, unknown>;
  /**
   * A simplified object that can be used for the DimensionBar interface
   */
  barUnits: Record<string, string | number>;
  /**
   * The title of the code segment
   */
  title: string;
  /**
   * The suffix to display after the unit
   */
  suffix?: string;
  /**
   * Source example
   */
  sourceExample: string;
  /**
   * The name of the object
   */
  objectName: string;
};

export function DimensionViewer({
  units,
  title,
  sourceExample,
  objectName,
  barUnits,
  suffix = "",
}: DimensionViewerProps): JSX.Element {
  return (
    <div data-testid={DimensionViewer.name}>
      <ObjectViewer object={units} title={objectName} />
      <p>
        To use these values import the <strong>{objectName}</strong> object and
        reference the required key.
      </p>
      <Source code={sourceExample} dark={true} language="css" />
      <DimensionBarWrapper title={title}>
        {Object.keys(barUnits).map((key, index) => {
          return (
            <DimensionBar
              key={`${title}${index}`}
              data-testid={DimensionViewer.name}
              width={`${barUnits[key]}${suffix}`}
            >
              <DimensionBarLabel>
                {barUnits[key]}
                {suffix}
                <br />({key})
              </DimensionBarLabel>
            </DimensionBar>
          );
        })}
      </DimensionBarWrapper>
    </div>
  );
}

const DimensionBarWrapper = ({ children }: HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      overflow: "scroll",
      width: "100%",
    }}
  >
    {children}
  </div>
);

const DimensionBar = ({
  children,
  width,
}: { width: string } & HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      background: colors.greenGrass,
      color: "white",
      flex: 1,
      marginBottom: "8px",
      width,
    }}
  >
    {children}
  </div>
);

const DimensionBarLabel = ({ children }: HTMLAttributes<HTMLSpanElement>) => (
  <span
    style={{
      display: "block",
      marginLeft: "8px",
    }}
  >
    {children}
  </span>
);
