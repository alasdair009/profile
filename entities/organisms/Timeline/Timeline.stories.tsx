import { Timeline } from "./Timeline";
import { Meta, StoryObj } from "@storybook/nextjs";
import jagexLogo from "../../assets/jagex-logo.svg";
import { expect, within } from "storybook/test";
import { TimelineElement } from "@/entities/organisms/Timeline/TimelineElement";

const meta: Meta<typeof Timeline> = {
  component: Timeline,
  args: {
    entries: [
      {
        date: "2011 - present",
        icon: jagexLogo,
        alt: "first logo",
        content: "abc",
      },
      {
        date: "2010",
        icon: jagexLogo,
        alt: "second logo",
        content: "def",
      },
      {
        date: "2009",
        icon: jagexLogo,
        alt: "third logo",
        content: "ghi",
      },
    ],
  },
  subcomponents: {
    // @ts-ignore
    TimelineElement: TimelineElement,
  },
};
export default meta;

export const Default: StoryObj<typeof Timeline> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const timelineElement = canvas.getByTestId(Timeline.name);
    const timelineEntries = canvas.getAllByTestId(TimelineElement.name);
    const firstTitle = args.entries[0].content as string;
    const firstDate = args.entries[0].date;
    const firstLogo = args.entries[0].alt;
    const lastTitle = args.entries[2].content as string;
    const lastDate = args.entries[2].date;
    const lastLogo = args.entries[2].alt;

    await expect(timelineElement).toBeInTheDocument();
    await expect(timelineEntries).toHaveLength(args.entries.length);
    await expect(canvas.getByText(firstTitle)).toBeInTheDocument();
    await expect(canvas.getByText(firstDate)).toBeInTheDocument();
    await expect(canvas.getByAltText(firstLogo)).toBeInTheDocument();
    await expect(canvas.getByText(lastTitle)).toBeInTheDocument();
    await expect(canvas.getByText(lastDate)).toBeInTheDocument();
    await expect(canvas.getByAltText(lastLogo)).toBeInTheDocument();
  },
};
