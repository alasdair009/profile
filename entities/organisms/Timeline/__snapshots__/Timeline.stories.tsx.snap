// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/Timeline Default smoke-test 1`] = `
<section class="sc-braxZu">
  <div class="vertical-timeline vertical-timeline--animate vertical-timeline--two-columns">
    <div class="sc-drnuxz vertical-timeline-element">
      <div id
           class="vertical-timeline-element"
      >
        <span class="vertical-timeline-element-icon bounce-in"
              style="background: rgb(46, 155, 38); color: rgb(255, 255, 255);"
        >
          <img alt
               loading="lazy"
               width="46"
               height="42"
               decoding="async"
               data-nimg="1"
               class="sc-hDcvty gtuqAY"
               srcset="/static/media/entities/assets/jagex-logo.svg?w=48&amp;q=75 1x, /static/media/entities/assets/jagex-logo.svg?w=96&amp;q=75 2x"
               src="/static/media/entities/assets/jagex-logo.svg?w=96&amp;q=75"
               style="color: transparent;"
          >
        </span>
        <div class="vertical-timeline-element-content bounce-in"
             style="background: rgb(46, 155, 38); color: rgb(255, 255, 255);"
        >
          <div class="vertical-timeline-element-content-arrow"
               style="border-right: 0.5rem solid rgb(46, 155, 38);"
          >
          </div>
          abc
          <span class="vertical-timeline-element-date">
            2011 - present
          </span>
        </div>
      </div>
    </div>
    <div class="sc-drnuxz vertical-timeline-element">
      <div id
           class="vertical-timeline-element"
      >
        <span class="vertical-timeline-element-icon bounce-in"
              style="background: rgb(46, 155, 38); color: rgb(255, 255, 255);"
        >
          <img alt
               loading="lazy"
               width="46"
               height="42"
               decoding="async"
               data-nimg="1"
               class="sc-hDcvty gtuqAY"
               srcset="/static/media/entities/assets/jagex-logo.svg?w=48&amp;q=75 1x, /static/media/entities/assets/jagex-logo.svg?w=96&amp;q=75 2x"
               src="/static/media/entities/assets/jagex-logo.svg?w=96&amp;q=75"
               style="color: transparent;"
          >
        </span>
        <div class="vertical-timeline-element-content bounce-in"
             style="background: rgb(46, 155, 38); color: rgb(255, 255, 255);"
        >
          <div class="vertical-timeline-element-content-arrow"
               style="border-right: 0.5rem solid rgb(46, 155, 38);"
          >
          </div>
          abc
          <span class="vertical-timeline-element-date">
            2011
          </span>
        </div>
      </div>
    </div>
    <div class="sc-drnuxz vertical-timeline-element">
      <div id
           class="vertical-timeline-element"
      >
        <span class="vertical-timeline-element-icon bounce-in"
              style="background: rgb(46, 155, 38); color: rgb(255, 255, 255);"
        >
          <img alt
               loading="lazy"
               width="46"
               height="42"
               decoding="async"
               data-nimg="1"
               class="sc-hDcvty gtuqAY"
               srcset="/static/media/entities/assets/jagex-logo.svg?w=48&amp;q=75 1x, /static/media/entities/assets/jagex-logo.svg?w=96&amp;q=75 2x"
               src="/static/media/entities/assets/jagex-logo.svg?w=96&amp;q=75"
               style="color: transparent;"
          >
        </span>
        <div class="vertical-timeline-element-content bounce-in"
             style="background: rgb(46, 155, 38); color: rgb(255, 255, 255);"
        >
          <div class="vertical-timeline-element-content-arrow"
               style="border-right: 0.5rem solid rgb(46, 155, 38);"
          >
          </div>
          abc
          <span class="vertical-timeline-element-date">
            2011
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
